src/components/interface/capmgr/memmgr.h

```diff
@@ -10,4 +11,23 @@ int memmgr_shared_page_alloc(vaddr_t *pgaddr);
 int memmgr_shared_page_allocn(int num_pages, vaddr_t *pgaddr);
 int memmgr_shared_page_map(int id, vaddr_t *pgaddr);
 
+/* This magic number is double the tls size defined in RK */
+#define TLS_AREA_SIZE 32
+#define TLS_NUM_PAGES (round_up_to_page(TLS_AREA_SIZE * MAX_NUM_THREADS) / PAGE_SIZE)
+#define TLS_BASE_ADDR 0x70000000
+
+void *memmgr_tls_alloc(unsigned int dst_tid);
+void *_memmgr_tls_alloc_and_set(void *area);
+
+static void *
+memmgr_tls_alloc_and_set(void *area)
+{
+	void *addr = _memmgr_tls_alloc_and_set(area);
+
+	/* Set area within addr for this tid */
+	*(void **)addr = area;
```

Comment:
i think im missing something here. You are casting a void * to a  void ** an then dereferencing? is that not the same as just addr = area ?

Reply:
`area` is confusing here.. Do you mean `data` ?? 
@maloneya I think it's setting the passed `data` (here `area`) to the `TLS AREA` that is allocated by the memmgr. 
I don't see a utility of `_set` here.. 
Interface users get the address and they can just set the data there..

Reply:
@phanikishoreg I see no reason to leave the responsibility to set the data up to the user of the interface. I agree that data would be a much better name though!
@maloneya casting addr as a void double pointer, and dereferencing that makes addr point to area, it doesn't set the addresses to equal the same thing.

Reply:
The real purpose of set however is that cos_thd_mod is called and sets the content of the gs register with addr.

Reply:
If the real purpose of this API is to call `cos_thd_mod`, shouldn't we pass in the `thdid`?? Or is this only for the current thread that we intend to use?? 

We'd want to setup the TLS before running a thread right? So, I'd think that we also want to pass in the thdid..

Reply:
I do *not* want all components with thread capabilities to be able to set `gs` as it has *global* impacts across multiple components.  This is clearly the responsibility of a shared manager component.  In the future, we will allow only a subset of operations on specific thread capabilities, and the only component that should be able to set `gs` is the capmgr (in this case).